src/proc/rotation-filter.cpp

```diff
@@ -98,89 +106,119 @@ namespace librealsense {
             tgt_intrin.fx = src_intrin.fy;
             tgt_intrin.fy = src_intrin.fx;
-
             tgt_intrin.ppx = src_intrin.height - src_intrin.ppy;
             tgt_intrin.ppy = src_intrin.ppx;
         }
         else if( value == -90 )
         {
-            _rotated_width = src_intrin.height;
-            _rotated_height = src_intrin.width;
-
+            rotated_width = src_intrin.height;
+            rotated_height = src_intrin.width;
             tgt_intrin.fx = src_intrin.fy;
             tgt_intrin.fy = src_intrin.fx;
-
             tgt_intrin.ppx = src_intrin.ppy;
             tgt_intrin.ppy = src_intrin.width - src_intrin.ppx;
         }
-        else if( value == 180 )  // 180 degrees rotation
+        else if( value == 180 )
         {
-            _rotated_width = src_intrin.width;
-            _rotated_height = src_intrin.height;
-
+            rotated_width = src_intrin.width;
+            rotated_height = src_intrin.height;
             tgt_intrin.fx = src_intrin.fx;
             tgt_intrin.fy = src_intrin.fy;
             tgt_intrin.ppx = src_intrin.width - src_intrin.ppx;
             tgt_intrin.ppy = src_intrin.height - src_intrin.ppy;
         }
         else { throw std::invalid_argument( "Unsupported rotation angle" ); }
 
-        tgt_intrin.width = _rotated_width;
-        tgt_intrin.height = _rotated_height;
+        // Update the per-stream rotated dimensions.
+        _rotated_dimensions[stream_key] = { rotated_width, rotated_height };
+
+        // Update dimensions for the intrinsics.
+        tgt_intrin.width = rotated_width;
+        tgt_intrin.height = rotated_height;
 
         tgt_vspi->set_intrinsics( [tgt_intrin]() { return tgt_intrin; } );
-        tgt_vspi->set_dims( _rotated_width, _rotated_height );
-        _last_rotation_value = value;
+        tgt_vspi->set_dims( rotated_width, rotated_height );
+
+        _last_rotation_values[stream_key] = value;
+        _target_stream_profiles[stream_key] = target_profile;
     }
 
-    rs2::frame rotation_filter::prepare_target_frame(const rs2::frame& f, const rs2::frame_source& source, rs2_extension tgt_type)
+    rs2::frame rotation_filter::prepare_target_frame( const rs2::frame & f,
+                                                      const rs2::frame_source & source,
+                                                      const rs2::stream_profile & target_profile,
+                                                      rs2_extension tgt_type )
     {
         auto vf = f.as<rs2::video_frame>();
-        auto ret = source.allocate_video_frame(_target_stream_profile, f,
-            vf.get_bytes_per_pixel(),
-            _rotated_width,
-            _rotated_height,
-            _rotated_width * vf.get_bytes_per_pixel(),
-            tgt_type);
+        std::pair< rs2_stream, int > stream_key( f.get_profile().stream_type(), f.get_profile().stream_index() );
+
+        if( _rotated_dimensions.find( stream_key ) == _rotated_dimensions.end() )
```

Comment:
On which case will this happen?

Reply:
Should not happen but I think it's OK to test.
There are cases where say `update_output_profile` returns without setting `_rotated_dimensions`, better to make sure we are aligned.

Reply:
@noacoohen, now that I think of it, do we really need `_rotated_dimensions`? I think the information is stored in `_target_stream_profiles`

Reply:
@OhadMeir removed it.